### PR TITLE
Receive invocation context from Proxy, pipe this into remote logging

### PR DESCRIPTION
**Pull request by dkolas**
This gives the remote logs the plugin/app handle, instance handle, version handle, as well as spaceId, tenantId, userId.

It also undoes the creation of the self.loggers for all apps and plugins, so we return to just using logging.x directly.

### CODE REVIEW

File: src/steamship/app/lambda_handler.py [hunk 1]
```diff
@@ -132,19 +138,27 @@ def handler(event: Dict, context: Dict = None) -> dict:
                 "type": "%(levelname)s",
                 "stack_trace": "%(exc_text)s",
                 "component": "app-plugin-lambda",
+                "userId": invocation_context.userId,
```
Comment (eob) [inline src/steamship/app/lambda_handler.py:141]: <does an excited dance.gif>

**Review by eob — DISMISSED** [review]
This looks great! 

I think the reversion back to just regular logging looks so much more natural to use

File: src/steamship/app/lambda_handler.py [hunk 2]
```diff
@@ -132,19 +138,27 @@ def handler(event: Dict, context: Dict = None) -> dict:
...
+                "spaceId": invocation_context.spaceId,
+                "tenantId": invocation_context.tenantId,
+                "invocableHandle": invocation_context.invocableHandle,
+                "invocableVersionHandle": invocation_context.invocableVersionHandle,
+                "invocableType": invocation_context.invocableType,
+                "path": event.get("invocation").get("appPath")
```
[the remark this answers is not in the document]
Comment (EniasCailliau) [reply src/steamship/app/lambda_handler.py:147]: With the default `{}` we can get rid of the if else below.

File: src/steamship/app/lambda_handler.py [hunk 3]
```diff
@@ -132,19 +138,27 @@ def handler(event: Dict, context: Dict = None) -> dict:
...
+                if event.get("invocation") is not None
+                else None,
```
Comment (EniasCailliau) [inline src/steamship/app/lambda_handler.py:149]:
```suggestion
```

File: src/steamship/app/request.py [hunk 4]
```diff
@@ -43,6 +43,17 @@ class LoggingConfig(BaseModel):
     loggingPort: str = None
 
 
+class InvocationContext(BaseModel):
+    tenantId: str = None
```
Comment (EniasCailliau) [inline src/steamship/app/request.py:47]: Can you turn this into a CamelModel please with camel_case properties please 😍

File: src/steamship/app/request.py [hunk 5]
```diff
@@ -43,6 +43,17 @@ class LoggingConfig(BaseModel):
...
+    userId: str = None
+    spaceId: str = None
+
+    invocableHandle: str = None
+    invocableVersionHandle: str = None
+    invocableInstanceHandle: str = None
+    invocableType: str = None
+
+
 class Request(BaseModel):
```
Comment (EniasCailliau) [inline src/steamship/app/request.py:57]: Can you turn this into a CamelModel please 🤩

File: src/steamship/plugin/service.py [hunk 6]
```diff
@@ -100,15 +100,15 @@ def model_cls(self) -> Type[TrainableModel]:
 
     def run(self, request: PluginRequest[T]) -> Union[U, Response[U]]:
         """Loads the trainable model before passing the request to the `run_with_model` handler on the subclass."""
-        self.logger.info("TrainablePluginService:run() - Loading model")
+        logging.info("TrainablePluginService:run() - Loading model")
```
Comment (EniasCailliau) [inline src/steamship/plugin/service.py:103]: I love how detailed your logging statement is, this makes debugging a breeze! I am afraid that we won't always be so detailed. 

Would it be an option to upgrade the logging format to include part of this automatically? 

An example:
```
FORMAT = "[%(filename)s:%(lineno)s - %(funcName)s] %(message)s"
```

Comment (dkolas) [reply src/steamship/plugin/service.py:103]: At least for remote logging, some of these fields are already in the structured part of the message (which makes it to structured fields in Elastic).  I'll add filename and lineno!

File: src/steamship/app/lambda_handler.py [hunk 7]
```diff
@@ -132,19 +138,27 @@ def handler(event: Dict, context: Dict = None) -> dict:
                 "type": "%(levelname)s",
                 "stack_trace": "%(exc_text)s",
                 "component": "app-plugin-lambda",
+                "userId": invocation_context.userId,
+                "spaceId": invocation_context.spaceId,
+                "tenantId": invocation_context.tenantId,
+                "invocableHandle": invocation_context.invocableHandle,
+                "invocableVersionHandle": invocation_context.invocableVersionHandle,
+                "invocableType": invocation_context.invocableType,
```
[the remark this answers is not in the document]
Comment (dkolas) [reply src/steamship/app/lambda_handler.py:146]: That's covered elsewhere - the fluentd container knows what env it's in, and injects it into all the log streams.

File: src/steamship/app/lambda_handler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkolas) [reply src/steamship/app/lambda_handler.py:147]: Thank you!  I'm not sure why I always forget that shortcut.

**Pull request merged**